ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt

```diff
@@ -54,12 +54,22 @@ internal class JsClientEngine(override val config: HttpClientEngineConfig) : Htt
         )
     }
 
-    // Adding "_capturingHack" to reduce chances of JS IR backend to rename variable,
-    // so it can be accessed inside js("") function
-    private fun createWebSocket(urlString_capturingHack: String): WebSocket {
+    @Suppress(
+        // Adding "_capturingHack" to reduce chances of JS IR backend to rename variable,
+        // so it can be accessed inside js("") function
+        "LocalVariableName",
+        // used in js() calls
+        "UNUSED_VARIABLE", "UNUSED_PARAMETER",
+    )
+    private fun createWebSocket(urlString_capturingHack: String, headers: Headers): WebSocket {
         return if (PlatformUtils.IS_NODE) {
             val ws_capturingHack = js("require('ws')")
-            js("new ws_capturingHack(urlString_capturingHack)")
+            val headers_capturingHack: dynamic = object {}
+            headers.forEach { name, values ->
+                val value = values.first()
+                headers_capturingHack[name] = value
```

Comment:
How does `first()` work with a header having multiple values, eg 
```
Cookie: name=value; name2=value2; name3=value3
```

Reply:
the header property is a JS object. And afaik you can't set multiple properties with the same name of a JS object, right?
So I assume the WebSocket package can't do that?

Reply:
Right, and the developers of NodeJS had the same problem: https://github.com/nodejs/node/issues/3591. According to https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2, they joined the values using a comma.
So I would use
```kotlin
val value = values.joinToString(",")
```` 
instead. But maybe, as I am not a webserver/RFC profi, I am wrong :)

Reply:
Well I am not a webserver/RFC profi either so IDK

Reply:
One last example: https://developer.mozilla.org/en-US/docs/Web/API/Headers/append#example. Mozilla also concats the values with a comma, if the header allows multiple values...

https://github.com/ktorio/ktor/blob/21771311f3dd1723d557ab50b5f1e2621d2d47e3/ktor-client/ktor-client-core/common/src/io/ktor/client/engine/Utils.kt#L37

Reply:
I added another unit test and it passes with your change so it seems to work